### DISCUSSION TITLE
Production Build Error on Vercel: 'act(...) is not supported

### DISCUSSION THREAD

**Discussion by bbtools-ps** [vercel/community#4225] [Help]
### Summary

I'm encountering an issue with my Next.js app when trying to build and deploy it on Vercel. The build process fails with the following error message:

`act(...) is not supported in production builds of React`

It's important to note that the same build command is successful both locally and when deployed on Netlify. The modified `build` command I'm using to run tests before the `next build` is:

`vitest run --reporter=verbose && next build`

Here's a breakdown of the situation:

- The error occurs specifically during the production build process on Vercel.
- The build command successfully runs tests locally and on Netlify, but it encounters issues on Vercel.
- The application appears to be behaving as expected in development mode.

I've already reviewed my code, dependencies, and environment variables, but I haven't been able to pinpoint the exact cause of the error in the Vercel production environment. Any guidance or insights on resolving this issue would be greatly appreciated.

Thank you in advance for your help!

### Steps to Reproduce

1. Modify the `build` script in `package.json` file to be: `vitest run --reporter=verbose && next build`
2. `npm run build`

**Comment by jose-guilherme93** [comment]
I have the same problem... with no solution

**Comment by Felt-Gaming-Labs** [comment]
Check for act() Usage: If act() is used explicitly in your tests, make sure it’s not leaking into the production build.
